src/core/net/srp_server.cpp

```diff
@@ -214,11 +214,41 @@ void Server::AddHost(Host *aHost)
     IgnoreError(mHosts.Add(*aHost));
 }
 
-void Server::RemoveAndFreeHost(Host *aHost)
+void Server::RemoveHost(Host *aHost, bool aRetainName, bool aNotifyServiceHandler)
 {
-    otLogInfoSrp("[server] fully remove host %s", aHost->GetFullName());
-    IgnoreError(mHosts.Remove(*aHost));
-    aHost->Free();
+    VerifyOrExit(aHost != nullptr);
```

Comment:
Under what scenarios is `aHost` equal to `nullptr`?

Reply:
The`RemoveService` allows `aService` to be nullptr and I think it would be expected that `RemoveHost` should also accept nullptr. I updated the usage of `RemoveHost` to avoid checking `aHost` before calling it.